flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/CheckpointBarrier.java

```diff
@@ -47,11 +48,18 @@
 	private final long id;
 	private final long timestamp;
 	private final CheckpointOptions checkpointOptions;
+	private final boolean isExactlyOnceMode;
 
+	@VisibleForTesting
 	public CheckpointBarrier(long id, long timestamp, CheckpointOptions checkpointOptions) {
+		this(id, timestamp, checkpointOptions, true);
+	}
+
+	public CheckpointBarrier(long id, long timestamp, CheckpointOptions checkpointOptions, boolean isExactlyOnceMode) {
```

Comment:
it would be better to make `isExactlyOnceMode` as an internal field of `CheckpointOptions`, then we can avoid touching the changes of multiple related classes.